frontends/ol-components/src/components/LearningResourceExpanded/LearningResourceExpanded.tsx

```diff
@@ -0,0 +1,524 @@
+      <ResourceDescription resource={resource} />
+    </Detail>
+  )
+}
+
+const ResourceTitle = ({ resource }: { resource?: LearningResource }) => {
+  if (!resource) {
+    return <Skeleton variant="text" height={20} width="66%" />
+  }
+  return (
+    <Typography variant="h5" component="h2">
+      {resource.title}
+    </Typography>
+  )
+}
+
+const ResourceDescription = ({ resource }: { resource?: LearningResource }) => {
+  if (!resource) {
+    return (
+      <>
+        <Skeleton variant="text" width="100%" />
+        <Skeleton variant="text" width="100%" />
+        <Skeleton variant="text" width="100%" />
+        <Skeleton variant="text" width="100%" />
+        <Skeleton variant="text" width="100%" />
+        <Skeleton variant="text" width="33%" />
+      </>
+    )
+  }
+  return (
+    <Description
+      dangerouslySetInnerHTML={{ __html: resource.description || "" }}
+    />
+  )
+}
+
+const TopicsSection: React.FC<{ topics?: LearningResourceTopic[] }> = ({
+  topics,
+}) => {
+  if (!topics) {
+    return null
+  }
+  return (
+    <Topics>
+      <Typography variant="subtitle2" component="h3">
+        Topics
+      </Typography>
+      <TopicsList>
+        {topics.map((topic) =>
+          topic.channel_url ? (
+            <ChipLink
+              key={topic.id}
+              size="medium"
+              label={topic.name}
+              href={topic.channel_url}
+              variant="outlined"
+            />
+          ) : (
+            <Chip
+              key={topic.id}
+              size="medium"
+              label={topic.name}
+              variant="outlined"
+            />
+          ),
+        )}
+      </TopicsList>
+    </Topics>
+  )
+}
+
+const InfoItem = ({
+  label,
+  Icon,
+  value,
+}: {
+  label: string
+  Icon: RemixiconComponentType
+  value: string | null
+}) => {
+  if (!value) {
+    return null
+  }
+  return (
+    <InfoItemContainer>
+      <Icon />
+      <InfoLabel>{label}</InfoLabel>
+      <InfoValue>{value}</InfoValue>
+    </InfoItemContainer>
+  )
+}
+
+const InfoSection = ({
+  run,
+  platformCode,
+}: {
+  run?: LearningResourceRun
+  platformCode?: PlatformEnum
+}) => {
+  if (!run) {
+    return null
+  }
+
+  const price = run.prices?.[0]
+  const displayPrice =
+    platformCode === PlatformEnum.Ocw || parseFloat(price!) === 0
+      ? "Free"
+      : price
+        ? `$${price}`
+        : null
+
+  const level = run.level?.[0]?.name || null
+
+  const instructors = run.instructors?.length
+    ? run.instructors
+        .filter((instructor) => instructor.full_name)
+        .map(({ full_name: name }) => name)
+        .join(", ")
+    : null
+
+  const languages = run.languages?.length
+    ? run.languages
+        .map((language) => ISO6391.getName(language.substring(0, 2)))
+        .join(", ")
+    : null
+
+  if (
+    [displayPrice, level, instructors, languages].every(
+      (element) => element === null,
+    )
+  ) {
+    return null
+  }
+
+  return (
```

Comment:
I'm noticing that Programs and Videos seem to be missing their metadata (e.g., duration). It's not handled in the InfoSection component.

<img width="150" alt="Screenshot 2024-05-30 at 11 19 01 AM" src="https://github.com/mitodl/mit-open/assets/9010790/20c71626-74b0-455d-bacc-6319a9e0d661">

Additionally, I think it would be really **good if we could add more metadata without modifying the component code** . I'm a little worried this component will become a big heap of if / else blocks.

IMO:
- what metadata gets shown should come from a config object
- as we add more metadata, we update the config object
- the component itself should be pretty simple

I'm thinking something like this:
```tsx
type InfoSectionProps = {
  resource: LearningResource
  run?: LearningResourceRun
}
const InfoSection = ({ resource, run }: InfoSectionProps) => {
  const infoItems = INFO_MAP[resource.resource_type]
    .map(({ label, value }) => ({ label, value: value(resource, run) }))
    .filter((x): x is { label: string; value: string } => !!x.value)
  
  if (infoItems.length === 0) return
  return (
    // render the items
  )
```

and a config object:
```ts

type InfoExtractor<T extends ResourceTypeEnum = ResourceTypeEnum> = (
  resource: LearningResource & { resource_type: T },
  run?: T extends
    | typeof ResourceTypeEnum.Course
    | typeof ResourceTypeEnum.Program
    ? LearningResourceRun
    : undefined,
) => string | null | undefined
type InfoMap = Record<
  ResourceTypeEnum,
  { label: string; value: InfoExtractor }[]
>

const extractPrice: InfoExtractor = (resource, run) =>
  resource.platform?.code === PlatformEnum.Ocw
    ? "Free"
    : run?.prices?.[0] && `$${run?.prices?.[0]}`
const extractLevel: InfoExtractor = (resource, run) => run?.level?.[0]?.name
const extractInstructors: InfoExtractor = (resource, run) =>
  run?.instructors
    ?.filter((instructor) => instructor.full_name)
    .map(({ full_name: name }) => name)
    .join(", ")
const extractDuration = (resource: LearningResource) => {
  if (resource.resource_type === ResourceTypeEnum.Video) {
    return resource.video.duration
  }
  if (resource.resource_type === ResourceTypeEnum.PodcastEpisode) {
    return resource.podcast_episode.duration
  }
  return null
}

const INFO_MAP: InfoMap = {
  [ResourceTypeEnum.Course]: [
    { label: "Price", value: extractPrice },
    { label: "Level", value: extractLevel },
    { label: "Instructors", value: extractInstructors },
  ],
  [ResourceTypeEnum.Program]: [
    { label: "Price", value: extractPrice },
    { label: "Level", value: extractLevel },
    { label: "Instructors", value: extractInstructors },
  ],
  [ResourceTypeEnum.Video]: [{ label: "Duration", value: extractDuration }],
  [ResourceTypeEnum.PodcastEpisode]: [
    { label: "Duration", value: extractDuration },
  ],
  [ResourceTypeEnum.Podcast]: [],
  [ResourceTypeEnum.VideoPlaylist]: [],
  [ResourceTypeEnum.LearningPath]: [],
}
```

Reply:
@jonkafton Following up with what I said in Slack. Per Ferdi's comment:

> General note, probably not for this PR -- The theory here is that all the common metadata for all learning resources are displayed in the card. **In the drawer, we display all metadata**.

Might be better to do the config as an array. (I do still think it should be a config object, not handled in the component) so that if a resource has a price, we always show the price, if it has a certificate, we always show that info. If it's a podcast with a certificate...that's weird. Probably a data issue that should be addressed on the backend.

Reply:
That's pushed, following your suggestion for item config above @ChristopherChudzicki. We have an array of item config (label, icon and value selector function). All items are tried across learning resource types and will be displayed if they return a value. We can add discriminator logic for resource types (and anything else) inside the selector functions.

Adds duration, "offered by" and "number of courses". "Date Posted" is missing though - I couldn't match that on anything in the VideoResource schema.